haystack/modeling/evaluation/eval.py

```diff
@@ -70,17 +72,20 @@ def eval(
         for step, batch in enumerate(tqdm(self.data_loader, desc="Evaluating", mininterval=10)):
             batch = {key: batch[key].to(self.device) for key in batch}
 
-            with torch.no_grad():
+            model_type = type(model)
+            if isinstance(model, DataParallel) or isinstance(model, WrappedDataParallel):
+                model_type = type(model.module)
```

Comment:
To be _really_ nitpicky, this makes more sense as a classic if-else block:

```suggestion
            if isinstance(model, DataParallel) or isinstance(model, WrappedDataParallel):
                model_type = type(model.module)
            else:
                model_type = type(model)
```

Reply:
I've updated to use the if-else block and I also changed it so we use `isinstance` over `type` so these if statements will still work when classes inherit from `AdaptiveModel` or `BiAdaptiveModel`.